src/types/vendor/express.d.ts

```diff
@@ -37,6 +38,7 @@ declare module 'express-session' {
     user?: {
       _id: IUserSchema['_id']
     }
+    sgid?: SgidUser
```

Comment:
should we set this in `setProfile` in `auth-sgid.controller`? don't think this is being used currently (lmk if I missed it out somewhere!)

Reply:
This is used in `setProfile` to verify if the selected profile is valid on [Line 220](https://github.com/opengovsg/FormSG/pull/6763/files#diff-1f3c9626dd630dc0f340b56cb7ebedabd983255deb8817f48369b3d3c321fe7cR220).

The actual "user" identity is stored under `session.user._id`, while the sgid profile information is stored under `session.sgid.profiles`.

Lmk if it's confusing.

Reply:
To be verbose.

1. `sgid.profiles` are valid profiles (can be multiple) that the user can assume. But the user can only assume 1 profile at any single time (identified through `user._id`.

2. I'm storing the profiles under `session` as it's a two-step process where we have a (1) callback from sgid supplying us the valid sgid profiles, and (2) where the user selects from the list of valid profiles.

Reply:
Ok yes, got it! I was wondering where the profiles were being set, realise it's in `getProfiles`